File: src/Modules/Metrics/ProjectManagement.Metrics.Domain/Enums/MetricEventType.cs
namespace ProjectManagement.Metrics.Domain.Enums;

public static class MetricEventType
{
    public const string PredictiveOverride = "predictive_override";
    public const string ForecastProactive  = "forecast_proactive";
    public const string SuggestionAccept   = "suggestion_accept";
    public const string SuggestionOverride = "suggestion_override";

    private static readonly HashSet<string> _valid =
    [
        PredictiveOverride, ForecastProactive,
        SuggestionAccept, SuggestionOverride
    ];

    public static bool IsValid(string type) => _valid.Contains(type);
}

File: frontend/project-management-web/src/app/core/models/metric-event-type.ts
export const MetricEventType = {
  PredictiveOverride: 'predictive_override',
  ForecastProactive: 'forecast_proactive',
  SuggestionAccept: 'suggestion_accept',
  SuggestionOverride: 'suggestion_override',
} as const;

export type MetricEventType = (typeof MetricEventType)[keyof typeof MetricEventType];
